README.md

```diff
@@ -187,6 +187,21 @@ Note, in Postgres streaming is going to be slower than non-streaming options due
 
 Streaming support is only implemented in the postgres backend at the moment, PRs welcome to add to other backends.
 
+## Prepared Statements
+See [benchmark mini_sql](https://github.com/discourse/mini_sql/tree/master/bench/prepared_perf.rb)
+[benchmark mini_sql vs rails](https://github.com/discourse/mini_sql/tree/master/bench/bilder_perf.rb).
+
+By default prepared cache size is 500 queries. Use prepared queries only for frequent queries.
+
+```ruby
+conn.prepared.query("select * from table where id = ?", id: 10)
+
+ids = (1..100).to_a
+builder = conn.build("select * from table /*where*/")
+builder.where("id IN (?)", ids)
+builder.prepared(ids.size <= 3).query
```

Comment:
I am not sure I understand this example ... why is `ids.size <= 3)` here?

Reply:
thoughtless use prepared statements can increase DB memory consumption and not provide much speed advantage

https://github.com/rails/rails/issues/14645

for example, we have a multi-select in the application, and we have statistics on the use of the selection of values

50% of cases choose 1 value
30% of cases choose 2 value
15% of cases choose 3 values
2% of cases choose 4 values
...

optimally create only 3 prepared queries

```sql
select * from table where id in ($1)
select * from table where id in ($1, $2)
select * from table where id in ($1, $2, $3)
```